File: Server/src/ws-app/api/event-hub.ts
import eventHub from 'src/event-hub'

export default {
  emitTemplatesChanged() {
    // io.of(`/temp`).emit('change')
    eventHub.emit('ws-app:templates:changed')
  },

  emitUserReportsChanged(userId: string) {
    eventHub.emit(`ws-app:user:${userId}:reports:changed`)
  }
}

File: iOS/WeeklySpending/Sources/Backend/EventHub.swift
import Foundation
import Combine
import Common
import Backend_deprecated
@preconcurrency import SocketIO

extension EventHub {
    static let shared = EventHub(baseURL: API.shared.baseURL)
    
    var templatesChangeEvents: AsyncStream<Void> {
        events("ws-app:templates:changed")
    }
    
    func userReportsChangeEvents(userID: String) -> AsyncStream<Void> {
        events("ws-app:user:\(userID):reports:changed")
    }
}
